src/event/server/ResourcePackStackSendEvent.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\event\server;
+
+use pocketmine\event\Event;
+use pocketmine\network\mcpe\NetworkSession;
+use pocketmine\resourcepacks\ResourcePack;
+
+class ResourcePackStackSendEvent extends Event{
+
+	/**
+	 * @param ResourcePack[] $resourcePackEntries
+	 */
+	public function __construct(private readonly NetworkSession $session, private array $resourcePackEntries, private bool $resourcePacksRequired, private bool $forceServerResources){}
+
+	public function getSession() : NetworkSession{
```

Comment:
I don't see why we need to involve the NetworkSession here.

Reply:
There are no player objects at this stage of the login sequence

Reply:
See PlayerPreLoginEvent.